store/populations.go

```diff
@@ -471,3 +473,147 @@ func iterateSortPVs(pvs []*pb.PropertyValue, action func(i int, p, v string)) {
 		action(i, p, pvMap[p])
 	}
 }
+
+func addObsSeries(
+	key string,
+	cacheValue string,
+	result map[string]pb.ObsTimeSeries,
+	mmethod string,
+	mprop string,
+	statType string,
+) error {
+	parts := strings.Split(key, "^")
+	dcid := strings.TrimPrefix(parts[0], util.BtObsSeriesPrefix)
+	val, err := util.UnzipAndDecode(cacheValue)
+	if err != nil {
+		return err
+	}
+	pbData := pb.PopObsPlace{}
+	jsonpb.UnmarshalString(string(val), &pbData)
+	ts := pb.ObsTimeSeries{PlaceName: pbData.Name, Data: map[string]float64{}}
+	for _, obs := range pbData.Observations {
+		if obs.MeasurementMethod != mmethod {
+			continue
+		}
+		if obs.MeasuredProp != mprop {
+			continue
+		}
+		msg := proto.MessageReflect(obs)
+		fd := msg.Descriptor().Fields().ByJSONName(statType)
+		if msg.Has(fd) {
+			ts.Data[obs.ObservationDate] = msg.Get(fd).Float()
+		}
```

Comment:
Does this work? I thought the current ObsTimeSeries proto does not have these fields. Instead mmethod, mprop are part of the key.

Reply:
As discussed, I added all the observation fields here to check. Also using the StatisticalVariable proto here.